File: August2022/C++/FirstUniqueCharacterInAString.cpp
class Solution {
public:
    // TC: O(n)
    // SC: O(26) ~ O(1)
    int firstUniqChar(string s) {
        int letters[26] = {0};        
        char c;
        for (int i = 0; i < s.size(); i++) {
            c = s[i];
            letters[c - 'a']++;
        }
        
        for (int i = 0; i < s.size(); i++) {
            c = s[i];
            if (letters[c - 'a'] == 1) {
                return i;
            }
        }
        
        return -1;
    }
};

File: August2022/Java/FirstUniqueCharacterInAString.java

// @saorav21994
// TC : O(n)
// SC : O(256) = O(1)

class Solution {
    public int firstUniqChar(String s) {
        int hash[] = new int[26];
        int l = s.length();
        for (int i = 0; i < l; i++) {
            int al = (int)s.charAt(i) - 97;
            hash[al] += 1;
        }
        // for (int idx : hash) System.out.print(idx + " ");
        int res = -1;
        for (int i = 0; i < l; i++) {
            if (hash[(int)s.charAt(i) - 97] == 1) {
                res = i;
                break;
            }
        }
        return res;
    }
}
